Source/Core/DiscIO/VolumeGC.cpp

```diff
@@ -210,31 +210,48 @@ bool CVolumeGC::LoadBannerFile() const
 		return m_banner_file_type != BANNER_INVALID;
 
 	std::unique_ptr<IFileSystem> file_system(CreateFileSystem(this));
-	size_t file_size = (size_t)file_system->GetFileSize("opening.bnr");
-	if (file_size == BNR1_SIZE || file_size == BNR2_SIZE)
+	if (!file_system)
 	{
-		m_banner_file.resize(file_size);
-		file_system->ReadFile("opening.bnr", m_banner_file.data(), m_banner_file.size());
+		m_banner_file_type = BANNER_INVALID;
+		return false;
+	}
 
-		u32 bannerSignature = *(u32*)m_banner_file.data();
-		if (file_size == BNR1_SIZE && bannerSignature == 0x31524e42)        // "BNR1"
-		{
-			m_banner_file_type = BANNER_BNR1;
-		}
-		else if (file_size == BNR2_SIZE && bannerSignature == 0x32524e42)   // "BNR2"
-		{
-			m_banner_file_type = BANNER_BNR2;
-		}
-		else
-		{
-			m_banner_file_type = BANNER_INVALID;
-			WARN_LOG(DISCIO, "Invalid opening.bnr. Type: %0x Size: %0lx", bannerSignature, (unsigned long)file_size);
-		}
+	std::unique_ptr<IFileInfo> file_info = file_system->FindFileInfo("opening.bnr");
+	if (!file_info)
+	{
+		m_banner_file_type = BANNER_INVALID;
+		return false;
 	}
-	else
+
+	size_t file_size = (size_t)file_info->GetSize();
+	if (file_size != BNR1_SIZE && file_size != BNR2_SIZE)
 	{
 		m_banner_file_type = BANNER_INVALID;
 		WARN_LOG(DISCIO, "Invalid opening.bnr. Size: %0lx", (unsigned long)file_size);
+		return false;
+	}
+
+	m_banner_file.resize(file_size);
+	if (file_system->ReadFile(file_info.get(), m_banner_file.data(), m_banner_file.size()) != file_size)
```

Comment:
There is an advantage of getting the size from the file info manually here. If the file size is invalid, the function returns before ReadFile is called, meaning that we don't have to waste time and memory loading a file that potentially can be huge. Skipping the `m_banner_file.resize(file_size)` line and shortening the line after that is what such a function could simplify while not changing any behavior, and I don't think that's worth it.